### PR TITLE
Task104/update download docs

### PR DESCRIPTION
**Pull request by at7**
-- add documentation
-- fix create data for entry
-- update visibility constraints when downloading as logged in user

### CODE REVIEW

File: modules/Bio/EnsEMBL/G2P/Utils/Downloads.pm [hunk 1]
```diff
@@ -24,12 +25,26 @@ use base qw(Exporter);
...
 my $confidence_category_attribs = {};
 my $allelic_requirement_attribs = {};
 my $mutation_consequence_attribs = {};
 
+=head2 download_data
+  Arg [1]    : String $downloads_dir - download file to this directory
+  Arg [2]    : String $file_name - name of download file
+  Arg [3]    : String $registry_file - for connecting to the G2P database
+  Arg [4]    : Boolean $is_logged_in - stores if the user is logged in
+  Arg [5]    : Arrayref $user_panels - all panels that a user can edit
```
Comment (dglemos) [inline modules/Bio/EnsEMBL/G2P/Utils/Downloads.pm]: Now the $user_panels is not being used in the code.

Comment (at7) [reply modules/Bio/EnsEMBL/G2P/Utils/Downloads.pm]: You are right. I will remove the $user_panels and also update the gene2phenotype_app code.

File: modules/Bio/EnsEMBL/G2P/Utils/Downloads.pm [hunk 2]
```diff
@@ -74,43 +111,47 @@ sub download_data {
...
-    my ($gfd_id, $created) = @$row;
-    $gfd_create_dates->{$gfd_id} = $created;
+    my ($gfd_panel_id, $created) = @$row;
+    $gfd_panel_create_dates->{$gfd_panel_id} = $created;
   }
   $sth->finish();
 
+  # create where clause to decide which data to download
+  # default where clause: only download entries which are visible
   my $where = ($panel_name eq 'ALL') ? 'WHERE gfdp.is_visible = 1' : "WHERE a.value = '$panel_name' AND gfdp.is_visible = 1";
```
Comment (dglemos) [inline modules/Bio/EnsEMBL/G2P/Utils/Downloads.pm]: On my test, here the $panel_name is empty and the downloaded file is also empty.

Comment (at7) [reply modules/Bio/EnsEMBL/G2P/Utils/Downloads.pm]: Are you logged in? And which panel are you downloading?

Comment (at7) [reply modules/Bio/EnsEMBL/G2P/Utils/Downloads.pm]: Are you also using this PR https://github.com/EBI-G2P/gene2phenotype_app/pull/22?

Comment (dglemos) [reply modules/Bio/EnsEMBL/G2P/Utils/Downloads.pm]: I was missing that PR. Now it works!

File: modules/Bio/EnsEMBL/G2P/Utils/Downloads.pm [hunk 3]
```diff
@@ -54,16 +70,37 @@ sub download_data {
...
   $csv->eol ("\r\n");
 
+  # preload phenotypes, organs and publications for all GenomicFeatureDiseases in the database 
   my $gfd_attribute_tables = {
     phenotype => {sql => 'SELECT gfdp.genomic_feature_disease_id, p.stable_id FROM genomic_feature_disease_phenotype gfdp, phenotype p WHERE gfdp.phenotype_id = p.phenotype_id;'},
     organ => {sql => 'SELECT gfdo.genomic_feature_disease_id, o.name FROM genomic_feature_disease_organ gfdo, organ o WHERE gfdo.organ_id = o.organ_id'},
     publication => {sql => 'SELECT gfdp.genomic_feature_disease_id, p.pmid FROM genomic_feature_disease_publication gfdp, publication p WHERE gfdp.publication_id = p.publication_id;'},
   };
 
+  # $gfd_attributes = {
```
Comment (dglemos) [inline modules/Bio/EnsEMBL/G2P/Utils/Downloads.pm:103]: I like the example but maybe would be better to have the example after the $gfd_attributes is populated.

**Comment by at7** [comment]
merged into master and release/104

**Pull request merged**